### DISCUSSION TITLE
How to play voice messages within nchat?

### DISCUSSION THREAD

**Discussion by PJBer** [d99kris/nchat#186] [Q&A]
Is it possible to play a voice message from within nchat?
Currently I'm always downloading the file and playing it, in another terminal, with ogg123. But playing it from within the chat would be a great improvment.

**Comment by d99kris** [comment]
Hi @PJBer - it should be possible to select the message and press `CTRL-v` (OpenFile). By default nchat uses `xdg-open` (Linux) or `open` (macOS) to open the file, so for it to work properly a player needs to be registered for the (ogg) filetype. To make sure the system is set up properly it's a good idea to try `xdg-open` / `open` (depending on OS) with an ogg/oga file received/downloaded using nchat. If that works, then `CTRL-v` should work too.

Alternatively one can use a custom OpenFile command by configuring `attachment_open_command` (see README for details) - just take note that the command needs to handle all filetypes you want to view (so one could create a wrapper script which calls different viewers/players depending on filetype for example).

**Comment by PJBer** [comment]
Thanks a lot, seems xdg-open has no default application for the ogg-filetype configured. So I'll dive into some man pages and try to set it to ogg123.

**Reply by Disonantemus** [reply]
I have `ogg123` (**vorbis-tools** package), but to use `xdg-open`, you need to create a `*.desktop` file, associated to **ogg123** and then set association with `xdg-mime`, maybe you can create one using `/usr/share/applications/mpv.desktop` as template or just use `mpv` to open every audio/video file (or something more simple like `htop.desktop`):

Something like this should work in your case, if you have **mpv** installed:
`xdg-mime default mpv.desktop audio/ogg`

### To change any file associations:

xdg-open only looks which desktop environment you have, and then run:

    gnome-open
    gvfs-open
    xfce-open
    ... etc

- Identify MIME type for a specified file:

    $ file -i in.pdf
    in.pdf: application/pdf; charset=binary

    $ xdg-mime query filetype in.pdf
    application/pdf

- Search 4 possible .desktop openers:

    $ find /usr/share/applications/ -type f -iname *zathura*.desktop
    applications/org.pwmt.zathura-pdf-mupdf.desktop
    applications/org.pwmt.zathura.desktop

- Set new default association:

    $ xdg-mime default org.pwmt.zathura.desktop application/pdf

- Query default for specified MIME type [optional]:

    $ xdg-mime query default application/pdf
    org.pwmt.zathura.desktop

Note: this in Arch Linux.

**Reply by d99kris** [reply]
Thanks for sharing 👍
